### PR TITLE
Integrate rule execution into logic

### PR DESCRIPTION
**Pull request by tcgj**
- Completes #47

### CODE REVIEW

File: src/main/java/budgetbuddy/logic/rules/RuleProcessingUtil.java [hunk 1]
```diff
@@ -27,9 +67,12 @@ public static Object extractAttribute(Attribute attribute, Transaction txn) {
...
             return txn.getDescription();
-        case AMOUNT:
-            long sign = txn.getDirection().equals(Direction.IN) ? 1 : -1;
-            return sign * txn.getAmount().toLong();
+        case IN_AMOUNT:
+            return (txn.getDirection().equals(Direction.IN) ? 1 : -1)
+                    * (txn.getAmount().toLong() / 100);
+        case OUT_AMOUNT:
+            return (txn.getDirection().equals(Direction.OUT) ? 1 : -1)
+                    * (txn.getAmount().toLong() / 100);
```
Comment (kenneth-fung) [inline src/main/java/budgetbuddy/logic/rules/RuleProcessingUtil.java]: What's up with the `/ 100`? If you want to display the amount in dollars and cents, I think amount's `toString` does that already.

File: src/main/java/budgetbuddy/logic/rules/performable/SetCategoryExpression.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+     */
+    public SetCategoryExpression(Value value) {
+        super(value);
+    }
+
+    @Override
+    public void perform(Model model, Transaction txn) {
+        requireAllNonNull(model, txn);
+        Category toSet = new Category(value.toString());
+        System.out.println("ADDED CATEGORY!!! " + toSet.toString());
```
Comment (kenneth-fung) [inline src/main/java/budgetbuddy/logic/rules/performable/SetCategoryExpression.java]: Might want to add a `todo` here. But up to you, so long as you remember.

Comment (tcgj) [reply src/main/java/budgetbuddy/logic/rules/performable/SetCategoryExpression.java]: Ah thanks for reminder

File: src/main/java/budgetbuddy/logic/rules/RuleProcessingUtil.java (hunk 1 above)
Comment (tcgj) [reply src/main/java/budgetbuddy/logic/rules/RuleProcessingUtil.java]: I need the long value of it that's why. It's not for display but for calculation

Comment (kenneth-fung) [reply src/main/java/budgetbuddy/logic/rules/RuleProcessingUtil.java]: Ah alright. You might want to leave it in cents though, `long` doesn't support decimals.

Comment (tcgj) [reply src/main/java/budgetbuddy/logic/rules/RuleProcessingUtil.java]: oh crap. Thanks for letting me know. I'll convert to double then

File: src/main/java/budgetbuddy/logic/rules/RuleProcessingUtil.java [hunk 3]
```diff
@@ -39,18 +82,58 @@ public static Object extractAttribute(Attribute attribute, Transaction txn) {
...
-        case TYPE_DESC:
-        case TYPE_CATEGORY:
+        case TYPE_STRING:
             // don't have to handle since value already stored as string
             break;
-        case TYPE_AMOUNT:
+        case TYPE_NUMBER:
             try {
-                Long.parseLong(value.toString());
+                Double.parseDouble(value.toString());
```
Comment (thwnus) [inline src/main/java/budgetbuddy/logic/rules/RuleProcessingUtil.java]: Doing the comparison as a double.. oh no.

But.. you know what, screw it.

**Pull request merged**